notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/main/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImpl.kt

```diff
@@ -0,0 +1,205 @@
+                    listOf(initialTransactionDetails),
+                    listOf(initialTransaction.notaryKey)
+                ).first().first()
+            } else null
+
+            // Send the response back to the client
+            session.send(uniquenessResult.toNotarizationResponse(initialTransactionDetails.id, signature))
+        } catch (e: Exception) {
+            logger.warn("Error while processing request from client. Cause: $e ${e.stackTraceToString()}")
+
+            session.send(
+                NotarizationResponse(
+                    emptyList(),
+                    NotaryExceptionGeneral(
+                        "Error while processing request from client. " +
+                        "Please contract notary operator for further details."
+                    )
+                )
+            )
+        }
+    }
+
+    /**
+     * A helper function that constructs an instance of [NotaryTransactionDetails] from the given transaction.
+     */
+    @Suspendable
+    private fun getInitialTransactionDetail(initialTransaction: UtxoSignedTransaction): NotaryTransactionDetails {
+        return NotaryTransactionDetails(
+            initialTransaction.id,
+            initialTransaction.metadata,
+            initialTransaction.outputStateAndRefs.count(),
+            initialTransaction.timeWindow,
+            initialTransaction.inputStateRefs,
+            initialTransaction.referenceStateRefs,
+            initialTransaction.notaryName,
+            initialTransaction.notaryKey
+        )
+    }
+
+    @Suspendable
+    private fun validateTransactionNotaryAgainstCurrentNotary(txDetails: NotaryTransactionDetails) {
+        val currentNotaryContext = memberLookup
+            .myInfo()
+            .memberProvidedContext
+        val currentNotaryServiceName = currentNotaryContext
+            .parse(NOTARY_SERVICE_NAME, MemberX500Name::class.java)
+
+        require(currentNotaryServiceName == txDetails.notaryName) {
+            "Notary service on the transaction ${txDetails.notaryName} does not match the notary service represented" +
+                    " by this notary virtual node (${currentNotaryServiceName})"
+        }
+    }
+
+    /**
+     * A function that will verify the signatures of the given [filteredTransactionsAndSignatures].
+     */
+    @Suspendable
+    private fun verifySignatures(
+        notaryKey: PublicKey,
+        filteredTransactionsAndSignatures: List<FilteredTransactionAndSignatures>
+    ) {
+        filteredTransactionsAndSignatures.forEach { (filteredTransaction, signatures) ->
+            require(signatures.isNotEmpty()) { "No notary signatures were received" }
+            filteredTransaction.verify()
+            for (signature in signatures) {
+                transactionSignatureService.verifySignature(
+                    filteredTransaction.id,
+                    signature,
+                    notaryKey
+                )
+            }
```

Comment:
Shouldn't that be 
```suggestion
          signatures.any {
                transactionSignatureService.verifySignature(
                    filteredTransaction.id,
                    signature,
                    notaryKey
                )
            }
```
The reason being that we need to verify that the notary has signed it - however, if we receive a non-notary signature, is that really a reason to fail signature verification?

Reply:
Since `transactionSignatureService.verifySignature()` doesn't return boolean and `.any {}` want us to return boolean, I tweaked the code like this. Is this acceptable?
```kotlin
signatures.any {
                try {
                    transactionSignatureService.verifySignature(
                        filteredTransaction.id,
                        it,
                        notaryKey
                    )
                    true
                } catch (e: Exception) {
                    false
                }
            }
  ```

Reply:
@vlajos had a very good suggestion to do something like we do in the UtxoSignedTransaction:
https://github.com/corda/corda-runtime-os/blob/b6f31f8d6f5a6f5dab797dc504e1825fc9d17b96/components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/UtxoSignedTransactionImpl.kt#L194